File: BotBuilder DotNet/libraries/Microsoft.Bot.Schema/SharePoint/BasicCardParameters.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System;
using System.Collections.Generic;
using System.Text;
using Newtonsoft.Json;

namespace Microsoft.Bot.Schema.SharePoint
{
    /// <summary>
    /// SharePoint basic card parameters.
    /// </summary>
    public class BasicCardParameters : BaseCardParameters
    {
        /// <summary>
        /// Initializes a new instance of the <see cref="BasicCardParameters"/> class.
        /// </summary>
        public BasicCardParameters()
        {
            // Do nothing
        }

        /// <summary>
        /// Gets or Sets the primary text of type <see cref="string"/>.
        /// </summary>
        /// <value>This value is the primary text to display.</value>
        [JsonProperty(PropertyName = "primaryText")]
        public string PrimaryText { get; set; }
    }
}

File: BotBuilderNodeJs/libraries/botframework-schema/src/sharepoint/basicCardParameters.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
import { BaseCardParameters } from './baseCardParameters';

/**
 * Sharepoint basic card view parameters
 */
export class BasicCardParameters extends BaseCardParameters {
    private primaryText: string;

    /**
     * Initializes a new instance of the BasicCardParameters class
     */
    public BasicCardParameters() {
        // Do nothing
    }

    /**
     * Sets primary text property of type string
     */
    public set PrimaryText(primaryText: string) {
        this.primaryText = primaryText;
    }

    /**
     * Gets primary text property of type string
     */
    public get PrimaryText(): string {
        return this.primaryText;
    }
}